lizrd/core/misc.py

```diff
@@ -318,12 +318,9 @@ def resolve_activation_name(activation: str) -> torch.nn.Module:
         raise ValueError(f"Unrecognized activation: {activation}")
 
 
-def propagate_store(module: torch.nn.Module, store=None):
-    """
-    This function propagates the cache from the module to all its children.
-    """
-    if store is None:
-        store = dict()
-    module.store = store
-    for child in module.children():
-        propagate_store(child, store)
+def add_additional_attributes_to_layer(
```

Comment:
Ej, ale chyba powinniśmy mieć tutaj rekurencję jakąś, co nie? Bo nasza warstwa to może bć jakiś sequential czy inne podobne. Chyba Attention tak działa obecnie. I chyba cache chcemy mieć w każdej warstwie, co nie?

Reply:
W szczególności chciałbym to mieć żeby niedługo supportować odpowiednią maskę attention która attenduje tylko do "swoich" przykładów gdy mamy wiele przykładów w jednym.

Reply:
Przywracam wersję z propagate tak, jak ustaliliśmy